packages/xo-web/src/common/custom-input.js

```diff
@@ -0,0 +1,39 @@
+import decorate from 'apply-decorators'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { injectState, provideState } from 'reaclette'
+
+const ENTER_KEY_CODE = 13
+const ESCAPE_KEY_CODE = 27
+
+const CustomInput = decorate([
```

Comment:
The input is not custom. The name needs to tell what the component does, not your specific use case. So maybe `EphemeralInput` or `OneTimeInput`?